### PR TITLE
Replace tempfile by tempdir as dev_dep

### PR DESCRIPTION
**Pull request by CPerezz**
Resolves: #493

### CODE REVIEW

File: src/circuit.rs [hunk 1]
```diff
@@ -291,7 +291,7 @@ mod tests {
     fn test_full() -> Result<(), Error> {
         use std::fs::{self, File};
         use std::io::Write;
-        use tempdir::TempDir;
+        use tempfile::TempDir;
 
         let tmp = TempDir::new("plonk-keys-test-full")
```
Comment (ZER0) [inline src/circuit.rs:296]: As GA said, the dirname will be autogenerated, you don't need to specify one. You can just call `let tmp = TempDir::new()`

**Review by ZER0 — CHANGES_REQUESTED** [review]

**Review by ZER0 — APPROVED** [review]

**Comment by CPerezz** [comment]
We can't merge that @ZER0 .

`tempfile` reilies on `rand` with features `getrandom` which requires `std`. Even this is a dev-dependency, is confusing cargo and it pulls in std into the mix.. 

We should consider even to remove this dep maybe and just create the dirs by ourseleves under `/tmp` or something similar.

**Review by vlopes11 — APPROVED** [review]
LGTM

**Comment by CPerezz** [comment]
Closing this until we have a better solution.

**Pull request closed without merging**